pkgs/build-support/dotnet/add-nuget-deps/default.nix

```diff
@@ -90,7 +90,7 @@ attrs
 
           # this needs to be before TMPDIR is changed, so the output isn't deleted
           # if it uses mktemp
-          depsFile=$(realpath "''${1:-${lib.escapeShellArg defaultDepsFile}}")
+          depsFile=$(realpath ''${1:-${lib.escapeShellArg defaultDepsFile}})
```

Comment:
I think this will break if $1 has spaces in it.

Reply:
maybe `escapeShellArg` is not suitable for use in this context, but I'm not sure how to do this in a robust way, except by maybe using a conditional assignment.

Reply:
Ah, you're right. Based on the error reported, I had assumed `escapeShellArg` may add its own quotes. But it does not:

```
nix-repl> pkgs = import <nixpkgs> {}

nix-repl> pkgs.lib.escapeShellArg "foo"
"foo"

```

If we can't just access the variable by passing it to stdenv, perhaps we should use [`escapeShellArgs`](https://noogle.dev/f/lib/escapeShellArgs), [`toShellVar`](https://noogle.dev/f/lib/toShellVar), or [`toShellVars`](https://noogle.dev/f/lib/toShellVars)?

Reply:
> Defining all required vars at the top of the script using toShellVars would be my preference.

yeah, I think that would be the cleanest option

Reply:
> I think this will break if $1 has spaces in it.

No, the original code broke when `$1` had spaces in it.

Reply:
> No, the original code broke when $1 had spaces in it. The new code does not.

The old code broke when `defaultDepsFile` had spaces, right?  I still think the new code will break if $1 has spaces:

```
$1="a b"
realpath ${1:-default}
realpath a b
```

Which will give realpath two args.

Reply:
> > No, the original code broke when $1 had spaces in it. The new code does not.
> 
> The old code broke when `defaultDepsFile` had spaces, right? I still think the new code will break if $1 has spaces:
> 
> ```
> $1="a b"
> realpath ${1:-default}
> realpath a b
> ```
> 
> Which will give realpath two args.

Sorry, my bad. I had swapped `$1` and `defaultDepsFile` in my head. We do need to quote the variable dereference; maybe it would be enough to remove `lib.escapeShellArg` instead?

Reply:
> maybe it would be enough to remove lib.escapeShellArg instead?

I believe we still need to escape other non-space chars.

Let's just make this a two step process:
- assign `toShellVars { inherit defaultDepsFile; }`
- dereference `"''${1:-$defaultDepsFile}")`